Sources/Smithy/Logging/LogAgent.swift

```diff
@@ -39,80 +41,85 @@ public enum LogAgentLevel: String, Codable, CaseIterable {
 }
 
 public extension LogAgent {
-    internal static func currentModule(filePath: String = #file) -> String {
-        let utf8All = filePath.utf8
-        return filePath.utf8.lastIndex(of: UInt8(ascii: "/")).flatMap { lastSlash -> Substring? in
-            utf8All[..<lastSlash].lastIndex(of: UInt8(ascii: "/")).map { secondLastSlash -> Substring in
-                filePath[utf8All.index(after: secondLastSlash) ..< lastSlash]
-            }
-        }.map {
-            String($0)
-        } ?? "n/a"
-    }
 
     /// Log a message passing with the `.info` log level.
-    func info(_ message: String, file: String = #file, function: String = #function, line: UInt = #line) {
+    func info(_ message: String, file: String = #fileID, function: String = #function, line: UInt = #line) {
         self.log(level: .info,
                  message: message,
                  metadata: nil,
-                 source: Self.currentModule(),
+                 source: currentModule(fileID: file),
```

Comment:
All these log-level-specific functions are changed similarly.  We weren't passing the caller's source file into the function that parses the module, hence every time we logged, it showed that it originated from the logger's module, not the caller's.

Reply:
Also, `#fileID` is the proper path to use in modern Swift production code, not `#file`.

`#fileID` omits path components from the host machine, and makes the module easier to parse.

See https://www.swift.org/documentation/api-design-guidelines/#parameter-names .